@here/harp-mapview-decoder/lib/TileDataSource.ts

```diff
@@ -198,15 +205,15 @@ export class TileDataSource<TileType extends Tile = Tile> extends DataSource {
      * `styleSetName` property) is found in `theme`.
      * @override
      */
-    setTheme(theme: Theme, languages?: string[], styleSetName?: string): void {
+    setTheme(theme: Theme | FlatTheme, languages?: string[], styleSetName?: string): void {
```

Comment:
i'm not sure that the signature of this function is correct. This code above is assuming that the `theme` set to this data source is already fully resolved (e.g. all the extends are evaluated), which is not the case.
I would expect setTheme to be an `async` function, accept a url or a theme and resolve the theme using the ThemeLoader.

Reply:
also i would remove `styleSetName` and languages from the signature of this function.

Reply:
Will be added that in #1949

Reply:
I squashed the two PRs, so the resolving of the themes is done... as well as the removal of the styleSetName I had added...